syntax = "proto3";

package app;

// The Data Service definition.
service DataService
{
  rpc GetNumber(NumberRequest) returns (NumberReply) { }
  rpc GetString(StringRequest) returns (StringReply) { }
  rpc GetFile(FileRequest) returns (stream FileChunk) { }
}

// The Number Request containing the name of the number.
message NumberRequest
{
  string name = 1;
}

// The Number Reply containing the number.
message NumberReply
{
  int32 number = 1;
}

// The String Request containing the index of the string.
message StringRequest
{
  int32 index = 1;
}

// The String Reply containing the string.
message StringReply
{
  string string_ = 1;
}

// The File Request containing the name of the file.
message FileRequest
{
  string filename = 1;
}

// The File Chunk containing a part of the file.
message FileChunk
{
  bytes content = 1;
}
